File: factorization_acmp.cpp
#include <iostream>
#include <math.h>

using namespace std;

int main() {
	
	int n, sqrtN;
	cin >> n;

	sqrtN = sqrt(n);
	
	int muls[sqrtN+1] = {};

	for (int d = 2; d <= sqrtN; d++) {
	
		while (n % d == 0) {
			muls[d]++;
			n /= d;
		}
	
	}
	
	bool wasStart = false;
	for (int i = 2; i <= sqrtN; i++) {
		for (int j = 0; j < muls[i]; j++) {
			if (wasStart) {
				cout << '*' << i;
			}
			else {
				cout << i;
				wasStart = true;
			}
		}
	}

	if (n!=1) {
		if (wasStart) {
			cout << '*' << n;
		} else {
			cout << n;
		}
	}

}

File: factorization_acmp.py
n = int(input())

sqrtN = int(n ** 0.5)
muls= [0] * (sqrtN+1)
ans=''


d = 2
while d <= sqrtN:
    
    while n % d == 0:
        muls[d] += 1
        n //= d

    d += 1

for i in range(2, sqrtN+1):
    for j in range(muls[i]):
        ans += str(i) + '*'

if n == 1:
    print(ans[:-1])
else:
    print(ans + str(n))

